lib/improver/blending/weighted_blend.py

```diff
@@ -810,25 +688,113 @@ def weighted_mean(self, cube, weights):
+
+        Returns:
+            numpy.int64:
+                Forecast reference time point in units of input cube coordinate
+        """
+        frt_coord = input_cube.coord("forecast_reference_time")
+        frt_units = frt_coord.units.origin
+        if cycletime is None:
+            return np.max(frt_coord.points)
+        else:
+            frt_calendar = frt_coord.units.calendar
+            cycletime_point = cycletime_to_number(
+                    cycletime, time_unit=frt_units, calendar=frt_calendar)
+            return np.round(cycletime_point).astype(np.int64)
+
+    def _set_coords_to_remove(self, input_cube):
+        """
+        Generate a list of coordinate names associated with the blend
+        dimension.  Unless these are time-related coordinates, they should be
+        removed after blending.
+
+        Args:
+            input_cube (iris.cube.Cube):
+                Cube to be blended
+        """
+        time_coords = ["time", "forecast_reference_time", "forecast_period"]
+        blend_dim, = input_cube.coord_dims(self.blend_coord)
+        self.crds_to_remove = []
+        for coord in input_cube.coords():
+            if coord.name() in time_coords:
+                continue
+            if blend_dim in input_cube.coord_dims(coord):
+                self.crds_to_remove.append(coord.name())
+
+    def _set_forecast_reference_time_and_period(self, blended_cube):
+        """
+        For cycle and model blending, update the forecast reference time and
+        forecast period coordinate points to the single most recent value,
+        rather than the blended average, and remove any bounds from the
+        forecast reference time. Modifies cube in place.
+
+        Args:
+            blended_cube (iris.cube.Cube)
+        """
+        blended_cube.coord("forecast_reference_time").points = [
+            self.cycletime_point]
+        blended_cube.coord("forecast_reference_time").bounds = None
+        if blended_cube.coords("forecast_period"):
+            blended_cube.remove_coord("forecast_period")
+        new_forecast_period = forecast_period_coord(blended_cube)
+        time_dim = blended_cube.coord_dims("time")
+        blended_cube.add_aux_coord(new_forecast_period, data_dims=time_dim)
+
+    def _update_blended_metadata(self, blended_cube, attributes_dict):
+        """
+        Update metadata after blending:
+        - For cycle and model blending, set a single forecast reference time
+        and period using self.cycletime_point or the latest cube contributing
+        to the blend
+        - Remove scalar coordinates that were previously associated with the
+        blend dimension
+        - Update attributes as specified via process arguments
+        Modifies cube in place.
+
+        Args:
+            blended_cube (iris.cube.Cube)
+            attributes_dict (dict or None)
+        """
+        if self.blend_coord in ["forecast_reference_time", "model_id"]:
```

Comment:
This plugin is also used to collapse over realization (via the threshold `--coord_collapse` option), and there are tests also covering collapsing percentiles, although I don't know what the use case for that is.  Hence the need to remove unneeded scalar coordinates relating to the blend dimension after blending.

Reply:
From a testing perspective, it would be useful if there was a unit test that "failed" this if test. As far as I can tell, all related unit tests always evaluate this if test as True, so I can't tell from the existing unit tests, whether `_set_forecast_reference_time_and_period` is actually doing something essential.

Reply:
I don't understand what you're asking.  `_set_forecast_reference_time_and_period` updates the blended forecast period and forecast reference time using the `cycletime` identified before blending.  This isn't a test and can't be failed.  There are several unit tests that check for these updates.  We don't want to update forecast time coordinates when collapsing over anything else.

I can show that these coordinates are not updated when collapsing realizations, if that is what you want?

Reply:
Yes, I think a tests that "shows that these coordinates are not updated when collapsing realizations" is essentially what is was suggesting.